src/db.rs

```diff
@@ -273,23 +423,57 @@ mod tests {
         assert_eq!(testpol.allowed_policies[0], 0);
         assert_eq!(testpol.min_fw_api_major, 23);
         assert_eq!(testpol.min_fw_api_minor, 0);
-        assert_eq!(testpol.allowed_build_ids[0], 10);
+        assert_eq!(testpol.allowed_build_ids[0], 0);
 
-        let _delsec = delete_polid(&sec).unwrap();
+        Ok(())
     }
 
     #[test]
-    fn test_secrets() {
-        let secid = "talking-donkies-can-fly".to_string();
-        let sec = "LXItLS4gIDEgcm9vdCByb290ICAxNzMgQXVnIDMwIDA2OjQ2IC52bWxpbnV6LTQuMTguMC0zMDUuMTcuMS5lbDhfNC54ODZfNjQuaG1hYwo=".to_string();
+    fn test_secrets() -> anyhow::Result<()> {
+        let secid = Uuid::new_v4().to_hyphenated().to_string();
+        let sec = Uuid::new_v4().to_hyphenated().to_string();
         let polid = 0;
-        let _insecid = insert_secret(&secid, &sec, polid).unwrap();
+        insert_secret(&secid, &sec, polid)?;
 
-        let tkey = get_secret(&secid).unwrap();
+        let tkey = get_secret(&secid)?;
 
         assert_eq!(tkey.id, secid);
         assert_eq!(tkey.payload, sec);
 
-        let _dsecid = delete_secret(&secid).unwrap();
+        delete_secret(&secid)?;
+        Ok(())
+    }
+
+    #[test]
+    fn test_insert_keyset() -> anyhow::Result<()> {
+        let keys: Vec<String> = vec![
+            "RGlyZSBXb2xmCg==".to_string(),
+            "VGhlIFJhY2UgaXMgT24K".into(),
+            "T2ggQmFiZSwgSXQgQWluJ3QgTm8gTGllCg==".into(),
+            "SXQgTXVzdCBIYXZlIEJlZW4gdGhlIFJvc2VzCg==".into(),
+            "RGFyayBIb2xsb3cK".into(),
+            "Q2hpbmEgRG9sbAo=".into(),
+            "QmVlbiBBbGwgQXJvdW5kIFRoaXMgV29ybGQK".into(),
+            "TW9ua2V5IGFuZCB0aGUgRW5naW5lZXIK".into(),
+            "SmFjay1BLVJvZQo=".into(),
+            "RGVlcCBFbGVtIEJsdWVzCg==".into(),
+            "Q2Fzc2lkeQo=".into(),
+            "VG8gTGF5IE1lIERvd24K".into(),
+            "Um9zYWxpZSBNY0ZhbGwK".into(),
+            "T24gdGhlIFJvYWQgQWdhaW4K".into(),
+            "QmlyZCBTb25nCg==".into(),
+            "UmlwcGxlCg==".into(),
+        ];
+        let ksetid = "UmVja29uaW5nCg==";
+        let polid = 1;
+        if let std::result::Result::Ok(_iksres) = insert_keyset(ksetid, &keys, polid) {
+            if let std::result::Result::Ok(ksids) = get_keyset_ids(&ksetid) {
```

Comment:
I think that if any of these fail the test will pass.
Maybe use `?` to return Error if anything fails, and write this sequentially.